public/video-ui/src/components/FormFields/TagPicker.js

```diff
@@ -34,359 +43,264 @@ export default class TagPicker extends React.Component {
     }
   }
 
-  parseTags = results => {
-    return results.map(result => {
-      if (result.type === TagTypes.keyword) {
-        let detailedTitle;
-
-        //Some webtitles on keyword tags are too unspecific and we need to add
-        //the section name to them to know what tags they are referring to
-
-        if (
-          result.webTitle !== result.sectionName &&
-          result.webTitle.split(' ').length <= 2
-        ) {
-          detailedTitle = result.webTitle + ' (' + result.sectionName + ')';
-        } else {
-          detailedTitle = result.webTitle;
-        }
+  fetchTags = searchText => {
 
-        return { id: result.id, webTitle: detailedTitle };
-      }
+     const tagTypes = this.props.tagType === TagTypes.keyword ? [TagTypes.series, TagTypes.keyword] : [this.props.tagType];
 
-      if (result.type === TagTypes.series) {
-        return { id: result.id, webTitle: result.webTitle + ' (series)' };
-      }
+    ContentApi.getTagsByType(searchText, tagTypes)
+      .then(capiResponses => {
+        const tags = capiResponses.reduce((tags, capiResponse) => {
+          return tags.concat(getTagDisplayNames(capiResponse.response.results));
+        }, []);
 
-      return { id: result.id, webTitle: result.webTitle };
-    });
-  };
+        this.setState({
+          capiTags: tags
+        });
+      })
+      .catch(() => {
+        this.setState({
+          capiTags: [],
+          capiUnavailable: true
+        });
+      });
+  }
 
   onUpdate = newValue => {
     this.setState({
       tagValue: newValue
     });
     this.props.onUpdateField(tagsToStringList(newValue));
+
+    this.setState({
+      capiTags: []
+    });
   };
 
-  getYoutubeInputValue = () => {
-    if (
-      this.state.tagValue.every(value => {
-        return value.id !== this.state.inputString;
-      })
-    ) {
-      return {
-        id: this.state.inputString,
-        webTitle: this.state.inputString
-      };
-    }
-    return [];
+  removeFn = (tag) => {
+    const newFieldValue = this.state.tagValue.filter(oldField => {
```

Comment:
Yes, `onUpdate` converts the list into a string before saving the value but the tagpicker handles the tags as lists.